src/LanguageServer/Program.cs

```diff
@@ -43,88 +43,77 @@ static int Main()
+                    string[] commandLineArguments = Environment.GetCommandLineArgs();
+                    if (commandLineArguments.Length == 2 && commandLineArguments[1] == "--probe")
+                    {
+                        // Give the language server a chance to start.
+                        await Task.Yield();
 
-                Task initializeTask = server.Initialize();
+                        // Show any exception encountered while starting the language server.
+                        if (initializeTask.IsFaulted || initializeTask.IsCanceled)
+                            await initializeTask;
 
-                // Special case for probing whether language server is startable given current runtime environment.
-                string[] commandLineArguments = Environment.GetCommandLineArgs();
-                if (commandLineArguments.Length == 2 && commandLineArguments[1] == "--probe")
-                {
-                    // Give the language server a chance to start.
-                    await Task.Yield();
+                        Console.Error.WriteLine("PROBE: Yes, the language server can start.");
 
-                    // Show any exception encountered while starting the language server.
-                    if (initializeTask.IsFaulted || initializeTask.IsCanceled)
-                        await initializeTask;
+                        return 0;
+                    }
 
-                    Console.Error.WriteLine("PROBE: Yes, the language server can start.");
+                    await initializeTask;
 
-                    return 0;
-                }
+                    log.Debug("Language server initialized by client.");
 
-                await initializeTask;
+                    if (server.Client.ProcessId != null)
+                    {
+                        terminator.Initialize(
+                            (int)server.Client.ProcessId.Value
+                        );
+                    }
 
-                log.Debug("Language server initialized by client.");
+                    await server.WasShutDown;
 
-                if (server.Client.ProcessId != null)
-                {
-                    terminator.Initialize(
-                        (int)server.Client.ProcessId.Value
-                    );
-                }
+                    log.Debug("Language server is shutting down...");
 
-                await server.WasShutDown;
+                    await server.WaitForExit;
 
-                log.Debug("Language server is shutting down...");
+                    log.Debug("Server has shut down. Preparing to terminate server process...");
 
-                await server.WaitForExit;
+                    log.Debug("Server process is ready to terminate.");
 
-                log.Debug("Server has shut down. Preparing to terminate server process...");
+                    return 0;
+                }
+            }
+            catch (AggregateException aggregateError)
```

Comment:
If we are fully async now then we no longer need to separately handle  `AggregateException`s (they come up when you call `GetResult()`).